Python/pywarpx/picmi.py

```diff
@@ -2599,16 +2599,35 @@ def diagnostic_initialize_inputs(self):
         else:
             species_names = [self.species.name]
 
+        # check if a different random fraction is given for each species
```

Comment:
Can the input parameter be a dictionary instead of a list? Something like `warpx_random_fraction = {electrons:0.4, ions:0.7}`.  The order of a list would depend on the order in which species are added which is something the could easily be modified in a complex input file, leaving this list inconsistent. Also, this would allow using default values - any species not specified in the dictionary would default to 1.

Reply:
Good idea on using a dictionary! I updated accordingly.